addon/mixins/model.js

```diff
@@ -55,13 +55,16 @@ export default Ember.Mixin.create({
     var _queryParamPropertyName = queryParamPropertyName(propertyName);
     var _queryIdPropertyName = queryIdPropertyName(propertyName);
     var currentQueryId = queryId++;
+
+    const oldParams = this.get(_queryParamPropertyName, params)
+
     this.set(_queryParamPropertyName, params);
     this.set(_queryIdPropertyName, currentQueryId);
 
     //get the relationship value, reloading if necessary
-    var value = this.reloadRelationship(propertyName);
+    var value = this.reloadRelationship(propertyName, JSON.stringify(params) === JSON.stringify(oldParams));
```

Comment:
Shouldn't this be `JSON.stringify(params) !== JSON.stringify(oldParams)` so that it force reloads the relationship if the params have changed?

Reply:
I believe that is correct...